server/src/@common/dto/user/user.dto.ts

```diff
@@ -1,52 +1,94 @@
-import { User } from '@prisma/client';
-import { appConfig } from 'config/config';
+import { Profile, User } from '@prisma/client';
+import { appConfig } from '../../../../config/config';
 import { ERole, EBan } from '../../enums/user.enum';
+import { ApiProperty, PartialType } from '@nestjs/swagger';
+import { IsDate, IsEnum, IsInt, IsISO31661Alpha2, IsOptional, IsString } from 'class-validator';
+import { Exclude } from 'class-transformer';
+import { IsSteamCommunityID } from '../../validators/is-steam-id.validator';
+import { ProfileDto } from './profile.dto';
 
 export class UserDto implements User {
+    @ApiProperty()
+    @IsInt()
     id: number;
+
+    @ApiProperty()
+    @IsSteamCommunityID()
     steamID: string;
+
+    @ApiProperty()
+    @IsString()
     alias: string;
-    aliasLocked: boolean;
-    private _avatar: string;
+
+    @ApiProperty()
+    @IsEnum(ERole)
     roles: ERole;
+
+    @ApiProperty()
+    @IsEnum(EBan)
     bans: EBan;
+
+    @ApiProperty()
+    @IsISO31661Alpha2()
     country: string;
+
+    @ApiProperty()
+    @IsDate()
     createdAt: Date;
+
+    @ApiProperty()
+    @IsDate()
     updatedAt: Date;
 
+    private _avatar: string;
+
+    @ApiProperty()
     get avatarURL(): string {
-        const bans = this.bans;
-        const isAvatarBanned = bans & (1 << 2); // TODO: Refactor however needed to use Ban 'enum' (cyclic dep issue occurs when requiring user model in this file)
-        if (isAvatarBanned) {
+        // TODO: apparently steam's hosting has changed and this stuff is on cloudfare now, ask goc
+        if (this.bans & EBan.BANNED_AVATAR) {
             return appConfig.baseURL + '/assets/images/blank_avatar.jpg';
         } else {
-            const avatar = this._avatar;
-            return avatar ? `https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/${avatar}` : null;
+            return this._avatar
+                ? `https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/${this.alias}`
+                : null;
```

Comment:
`https://avatars.cloudflare.steamstatic.com/${this._avatar}` (and not `this.alias`)